app/src/main/java/io/apicurio/registry/rest/v2/GroupsResourceImpl.java

```diff
@@ -564,6 +565,22 @@ public Response getArtifactVersion(String groupId, String artifactId, String ver
         return builder.build();
     }
 
+    /**
+     * @see io.apicurio.registry.rest.v2.GroupsResource#deleteArtifactVersion(java.lang.String, java.lang.String, java.lang.String)
+     */
+    @Override
+    public void deleteArtifactVersion(String groupId, String artifactId, String version) {
+        if (!restConfig.isArtifactVersionDeletionEnabled()) {
+            throw new NotFoundException("Artifact version deletion is not enabled.");
+        }
+
+        requireParameter("groupId", groupId);
+        requireParameter("artifactId", artifactId);
+        requireParameter("version", version);
+
+        storage.deleteArtifactVersion(gidOrNull(groupId), artifactId, version);
```

Comment:
We should think about and handle the following edge cases:
- What happens if the version is the latest version
- What happens if the version (or other versions) is in a `DISABLED` state
- What happens if the version is he only one in the artifact
  - We should require deleting the entire artifact instead

Reply:
+ tests for the edge cases

Reply:
Hi @jsenko, based on the code I see on AbstractSqlRegistryStorage#deleteArtifactVersion:
- If the latest version is deleted, then the latest version is re-assigned
![image](https://user-images.githubusercontent.com/34775979/216448694-37848e45-25b0-4a2c-bc9b-55084cc3a617.png)

- Looks like there's no state validation, meaning the latest can be re-assigned to a disabled version

- If artifact has only one version, then the whole artifact is deleted
![image](https://user-images.githubusercontent.com/34775979/216448891-abb198cc-d8bd-470c-826a-1a5504de1730.png)

This method has tests covered under AbstractRegistryStorageTest#testDeleteArtifactVersion. I see there is test that cover deleting the latest version:
![image](https://user-images.githubusercontent.com/34775979/216449188-edab5114-4003-4aa6-a6b9-862987950875.png)

I can help increasing test coverage for these edge cases and perhaps add some kind of validation at api level, but I am reluctant on changing the internal functionality itself as I am not fully aware of how this could impact other areas of the code - would be more efficient if these eventual internal functionality changes is handled by Apicurio team I believe. 

Please let me know your thoughts

Reply:
Btw, added some more coverage to AbstractSqlRegistryStorage#deleteArtifactVersion to help covering those edge-case scenarios